### PR TITLE
feat(plugin-chart-choropleth-map): scaffold and load map

### PR DESCRIPTION
**Pull request by kristw**
**NOTE: This is merging into a feature branch, not `master`, and will have subsequent PRs into the feature branch until completion.**

## 🏆 Enhancements

* Create `plugin-chart-choropleth-map` to replace the `country-map` and `world-map` charts.

### New features

* Support `topojson`, which has much smaller file size.
* Clean up the ambiguity of which field to use as a `key` for each polygon.
* Allow customizing projection and rotation for each map.
* Full `react` and `typescript`

### Coming in later PRs

* Encoding
* Zoom & pan
* Tooltip

### Screenshots

image
image

### CODE REVIEW

File: plugins/plugin-chart-choropleth-map/src/maps/index.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    url: world,
+    keyField: 'id',
+    projection: 'Equirectangular',
+    rotate: [-9, 0, 0],
+  },
+];
+
+const mapsLookup: Record<string, RawMapMetadata> = keyBy((map: RawMapMetadata) => map.key)(maps);
+
+export { mapsLookup, maps };
```
Comment (ktmud) [inline plugins/plugin-chart-choropleth-map/src/maps/index.ts:28]: Style nit:

```ts
export const maps: ...
export const mapsLookup: ...
```

File: plugins/plugin-chart-choropleth-map/src/maps/index.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  {
+    key: 'world',
+    name: 'World Map',
+    type: 'topojson',
+    url: world,
+    keyField: 'id',
+    projection: 'Equirectangular',
+    rotate: [-9, 0, 0],
+  },
+];
```
Comment (ktmud) [inline plugins/plugin-chart-choropleth-map/src/maps/index.ts:24]: How about

```ts
export const maps = {
  usa: {
    ...
  },
  world: {
    ...
  }
};
export const mapChoices = Object.keys(maps); // or manually curated list
```

You get duplicate key checks for free.

File: plugins/plugin-chart-choropleth-map/src/plugin/index.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { t } from '@superset-ui/translation';
+import { ChartMetadata, ChartPlugin } from '@superset-ui/chart';
+import transformProps from './transformProps';
+import thumbnail from '../images/thumbnail.png';
+
+const metadata = new ChartMetadata({
+  description: 'Choropleth Map',
```
Comment (ktmud) [inline plugins/plugin-chart-choropleth-map/src/plugin/index.ts:25]: `t('Choropleth Map')`?

File: plugins/plugin-chart-choropleth-map/src/chart/MapMetadata.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  config: RawMapMetadata;
+
+  keyAccessor: (...args: unknown[]) => string;
+
+  constructor(metadata: RawMapMetadata) {
+    const { keyField } = metadata;
+
+    this.config = metadata;
+    const keyAccessor = get(keyField);
+    this.keyAccessor = keyAccessor;
```
Comment (ktmud) [inline plugins/plugin-chart-choropleth-map/src/chart/MapMetadata.ts:17]: `this.keyAccessor = get(keyField)`

File: plugins/plugin-chart-choropleth-map/src/maps/index.ts (hunk 2 above)
Comment (kristw) [reply plugins/plugin-chart-choropleth-map/src/maps/index.ts]: 👍

**Pull request merged**